### ISSUE TITLE
StringName changes break project settings for older versions

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#39616]
**Godot version:**

02f7908

**Issue description:**

With the changes to StringNames in 4.0, some of the project settings are converted to the new format (e.g. class names). This results in parsing errors in older Godot versions, which makes the project being listed as `unnamed` and the settings are not recognized.

Now, theoretically this is expected due to breaking changes, but this makes `config_version` half-useless, because older versions can't read it (4.0 should bump the version btw). I wonder if we can handle this in more forward-compatible way.

**Comment by visionm45** [comment]
so I just built the new version of godot and for whatever reason it didn't work but what it did manage to do is break my project.  I removed all the @ symbols from the project file and changed a couple things in the main scene to get it to open but it won't run just throws errors.  

Basically my options to fix this seem to be that I need the new version to be released, I need to be able to correctly build the new version or I need to be able to somehow revert whatever the latest master branch of godot changed.

I do consider this a bug.
